### ISSUE TITLE
[cython] GeoSeries.reindex has wrong filling logic

### ISSUE THREAD

**Issue by jorisvandenbossche** [geopandas/geopandas#590]
```
In [2]: s = GeoSeries([Point(1,1), Point(2,2)])

In [3]: s.reindex([0,1,2])
Out[3]: I am densified (3 elements)

0    POINT (1 1)
1    POINT (2 2)
2    POINT (2 2)
dtype: object
```

while this should be `Point(1 1), Point(2 2), None`

The reason for this is because the `indexer` that is passed to `GeometryBlock.take_nd` (and currently thus `GeometryArray.take`) has the form of `array([ 0,  1, -1])`:

```
> /home/joris/scipy/geopandas/geopandas/_block.py(129)take_nd()
-> new_values = self.values[indexer]
(Pdb) l
127  	        # TODO implement take_nd on GeometryArray
128  	        # new_values = self.values.take_nd(indexer, fill_value=fill_value)
129  ->	        new_values = self.values[indexer]
130  	

(Pdb) indexer
array([ 0,  1, -1])
```

**Comment by jorisvandenbossche** [comment]
Actually, it is not using `GeometryArray.take`, but its `__getitem__`. And `take` already seems to support the '-1' indexer case:

https://github.com/geopandas/geopandas/blob/f9aa2202e47fdae4764fee20d15394d11a6d600e/geopandas/vectorized.pyx#L1208-L1211

should be an easy fix then!

**Comment by jorisvandenbossche** [comment]
Closed by  #591
